htmlhint-server/src/server.ts

```diff
@@ -60,6 +65,74 @@ function getRange(error: htmlhint.Error, lines: string[]): any {
     };
 }
 
+function loadRules(HTMLHint: any, force: boolean): any {
+    if (force) {
+        rulesLoaded = false;
+    }
+
+    if (rulesLoaded || htmlhint == null) {
+        return;
+    }
+    try {
+        if (!settings.htmlhint || !settings.htmlhint.rulesDir) {
+            return;
+        }
+        let rulesDir = settings.htmlhint.rulesDir;
+        let absoluteDir = '';
+        //check absolute path
+        let exists = fs.existsSync(rulesDir);
+        if (exists) {
+            absoluteDir = rulesDir;
+        } else {
+            //relative dir, find it based on workspace roots
+            absoluteDir = path.resolve(rootFolder, rulesDir);
+            exists = fs.existsSync(absoluteDir);
+        }
+
+        if (exists) {
+            //load all rules
+            loadCustomRules(absoluteDir, HTMLHint);
+        }
+    }
+    catch (e) { }
+    finally {
+        rulesLoaded = true;
+    }
+}
+
+// load custom rles
+function loadCustomRules(rulesdir:string, HTMLHint:any):any{
+    rulesdir = rulesdir.replace(/\\/g, '/');
+    if(fs.existsSync(rulesdir)){
+        if(fs.statSync(rulesdir).isDirectory()){
+            rulesdir += /\/$/.test(rulesdir)?'':'/';
+            rulesdir += '**/*.js';
+            var arrFiles = glob.sync(rulesdir, {
```

Comment:
A comment here explaining what files you're going to load would be helpful as well.